### PR TITLE
Slow and No Internet Handling

### PR DESCRIPTION
**Pull request by manthan-25**

### CODE REVIEW

File: ios/ReachabilityHandler.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import Foundation
+
+@objc(ReachabilityHandler)
+open class ReachabilityHandler: NSObject, ReachabilityObserverDelegate {
+    
+    fileprivate var isConnected: Bool
+    //MARK: Lifecycle
+    
+    required override public init() {
```
Comment (jitendralakhmani-proptiger) [inline ios/ReachabilityHandler.swift]: remove required and public

File: ios/ReachabilityHandler.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//  RNNetInfo
+//
+//  Created by Manthan Patel on 7/22/19.
+//  Copyright © 2019 Facebook. All rights reserved.
+//
+
+import Foundation
+
+@objc(ReachabilityHandler)
+open class ReachabilityHandler: NSObject, ReachabilityObserverDelegate {
```
Comment (jitendralakhmani-proptiger) [inline ios/ReachabilityHandler.swift]: open remove

File: android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    private void setConnectionStatus(boolean status) {
+        if (netInfo != null) {
+            netInfo.setNetConnected(status);
+        }
+    }
+
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/NetworkBroadcastReceiver.java:19]: put public methods above private

File: android/src/main/java/com/reactlibrary/NetworkConnection.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        Context context = mcontext.getApplicationContext();
+        ConnectivityManager cm =
+                (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        return (activeNetwork != null &&
+                activeNetwork.isConnected());
+    }
+
+    public static NetworkConnection getInstance(ReactApplicationContext... reactContext) {
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/NetworkConnection.java]: Single argument for ReactApplicationContext

File: android/src/main/java/com/reactlibrary/RNNetInfoModule.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+  private NetworkConnection netInfo;
+  private BroadcastReceiver mReceiver;
+  private boolean isReceiverRegistered = false;
+
+  public RNNetInfoModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    netInfo = NetworkConnection.getInstance(reactContext);
+    mReceiver = new NetworkBroadcastReceiver();
+    getReactApplicationContext().addLifecycleEventListener(this);
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/RNNetInfoModule.java]: this should be done only after registration

File: android/src/main/java/com/reactlibrary/RNNetInfoModule.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      } catch (Exception e) {
+          callback.reject(e);
+          return false;
+      }
+    }
+
+    private void registerReceiverIfNecessary(BroadcastReceiver receiver) {
+        if (getCurrentActivity() == null) return;
+        try {
+            getCurrentActivity().registerReceiver(
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/RNNetInfoModule.java:65]: No need to do this ifRegistered is true.

File: android/src/main/java/com/reactlibrary/NetworkConnection.java [hunk 7]
```diff
@@ -29,9 +29,9 @@ public static boolean updateConnectionFlag() {
                 activeNetwork.isConnected());
     }
 
-    public static NetworkConnection getInstance(ReactApplicationContext... reactContext) {
-        if (netInfo == null && reactContext.length >= 1) {
-            netInfo = new NetworkConnection(reactContext[0]);
+    public static NetworkConnection getInstance(ReactApplicationContext reactContext) {
```
Comment (Sunil18) [inline android/src/main/java/com/reactlibrary/NetworkConnection.java]: Synchronised

**Review by Sunil18 — APPROVED** [review]

**Pull request merged**